### PR TITLE
Replace tag selection window with a search string #48

### PR DESCRIPTION
**Pull request by bersegosx**

### CODE REVIEW

File: static/app/controllers/triggers.ts [hunk 1]
```diff
@@ -138,8 +139,11 @@ export class TriggersController {
 				tag.data = data;
 			});
 		} else {
-			if (!this.$scope.tags_filter.selection.contains(tag) && this.$scope.tags.contains(tag)) {
-				this.$scope.tags_filter.selection.push(tag);
+			if (this.$scope.search_string === undefined) {
+				this.$scope.search_string = "";
+			}
+			if (this.$scope.search_string.indexOf('#' + tag.value) == -1) {
+				this.$scope.search_string += ' #' + tag.value;
```
Comment (beevee) [inline static/app/controllers/triggers.ts:147]: Can you explain this, please?

**Comment by beevee** [comment]
Seems like you "fixed" a test by simply deleting it. What about your new functionality? Does it require a test?

Also, seems like you've lost an existing functionality of saving selected tags to cookies.

**Comment by beevee** [comment]
Is it possible to put the search bar in the header, between Moira logo and buttons on the right?

**Comment by beevee** [comment]
I also noticed that you send an immediate API request for each keypress without any delay. This can put significant load on our backend. I don't think it's a good practice.

File: static/app/controllers/triggers.ts (hunk 1 above)
Comment (bersegosx) [reply static/app/controllers/triggers.ts:147]: Лады, только тут вроде нечего рассказывать. При клике по тегу без зажатого `alt` вставляем тег в поисковую строку c #.

**Comment by bersegosx** [comment]
> Seems like you "fixed" a test by simply deleting it

Да нет же - https://github.com/moira-alert/web/pull/49/files#diff-6338df6ec8dacc4f6eccbeb31d1bccd9R39

> Is it possible to put the search bar in the header, between Moira logo and buttons on the right?

Возможно.

Фичу с сохранением тегов в куку верну, тесты и delay при поисковых запросах добавлю.

File: static/app/directives/search.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+export function Search(): ng.IDirective {
+
+	const max_items:number = 99;
+
+	return {
+		restrict: 'E',
+		template: require('./templates/search.html'),
+
+		replace: true,
+		transclude: true,
```
Comment (gmlexx) [inline static/app/directives/search.ts]: What are you going to transclude here?

Comment (bersegosx) [reply static/app/directives/search.ts]: Копипаста, поправил.

**Pull request closed without merging**